.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,134 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: "Ubuntu Benchmark"
+
+on:
+  pull_request:
+    paths:
+      - 'velox/**'
+      - 'third_party/**'
+      - '.github/workflows/benchmark.yml'
+  push:
+    branches: [main]
+
+permissions:
+  contents: read
+
+defaults:
+  run:
+    shell: bash
+#TODO  concurrency groups?
+jobs:
+  benchmark:
+    runs-on: 8-core
+    container: ghcr.io/facebookincubator/velox-dev:amd64-ubuntu-22.04-avx
+    env:
+      CCACHE_DIR: "${{ github.workspace }}/.ccache/"
+      CCACHE_BASEDIR: "${{ github.workspace }}"
+      BINARY_DIR: "${{ github.workspace }}/benchmarks/"
+      LINUX_DISTRO: "ubuntu"
+      RESULTS_ROOT: "${{ github.workspace }}/benchmark-results"
+      BASELINE_OUTPUT_PATH: "${{ github.workspace }}/benchmark-results/baseline/"
+      TARGET_OUTPUT_PATH: "${{ github.workspace }}/benchmark-results/target/"
+    steps:
+      - name: "Setup ccache and python"
+        run: |
+            # Set up ccache configs .
+            mkdir -p .ccache
+            ccache -sz -M 5Gi
+            
+            # We can not use setup-python as we are running in a container
+            # TODO add to docker image and remove here
+            apt update
+            apt install -y lsb-release python3 pip
+
+      - uses: actions/cache/restore@v3
+        id: restore-cache
+        with:
+          path: ".ccache"
+          key: ccache-benchmark-${{ github.sha }}
+          restore-keys: |
+            ccache-benchmark-
+
+      - name: "Checkout Base"
+        uses: actions/checkout@v3
+        with:
+          path: 'velox'
+          ref: ${{ github.event.pull_request.base.sha || github.sha }}
+          submodules: 'recursive'
+
+      - name: Build Base Benchmarks
+        working-directory: velox
+        run: |
+            n_cores=$(nproc)
+            make benchmarks-basic-build NUM_THREADS=$n_cores MAX_HIGH_MEM_JOBS=$((n_cores/2)) MAX_LINK_JOBS=$n_cores
+            ccache -s
+            mkdir -p ${BINARY_DIR}/baseline/
+            cp -r --verbose _build/release/velox/benchmarks/basic/* ${BINARY_DIR}/baseline/
+
+      - name: "Checkout Head"
+        if: ${{ github.event_name == 'pull_request' }}
+        uses: actions/checkout@v3
+        with:
+          clean: false
+          path: 'velox'
+          submodules: 'recursive'
+
+      - name: Build Target Benchmarks
+        if: ${{ github.event_name == 'pull_request' }}
+        working-directory: velox
+        run: |
+            n_cores=$(nproc)
+            make benchmarks-basic-build NUM_THREADS=$n_cores MAX_HIGH_MEM_JOBS=$n_cores MAX_LINK_JOBS=$n_cores
+            ccache -s
+            mkdir -p  ${BINARY_DIR}/target/
+            cp -r --verbose _build/release/velox/benchmarks/basic/*  ${BINARY_DIR}/target/
```

Comment:
Am I missing a section where we build the baseline (here, I believe, called `target`) for the merge-to-main scenario? These both look like they are only happening on pull requests, or am I missing something?

Reply:
Oh, actually I see what's going on now! The contender build for pushes to main are in the step above (labelled as "base") and then this is the section that builds the contender for PRs. 

HMMM that's really confusing IMO. Would it be possible to have the contender builds|benchmarks be in the same steps? Since ultimately we _do_ want both of those circumstances to send benchmarks up to conbench.

Reply:
Hm yes I was not really happy with copying the results over but I kind of got.stuck on having to build base first, which is of course not required... I will think of a clearer way.

Reply:
Additionally -- in the `merge-to-main` case we actually want to use `HEAD~1` as the baseline. (You can do this in a followup PR since it doesn't affect what's uploaded to Conbench if you wish)

Reply:
@austin3dickey I don't think it makes sense to run HEAD~1 on merge to main as HEAD~1 data already exists in conbench and the new data will not be uploaded and the combination was already run on the commit in the PR. As the PR was merged the devs likely looked at the results and  deemed them acceptable, so it seem superfluous to run them again.

Reply:
I changed it around to always build the contender and only build baseline on PRs :+1:

Reply:
>As the PR was merged the devs likely looked at the results and deemed them acceptable, so it seem superfluous to run them again.

Ah yeah good point. When we implement retries again we probably won't have to worry about doing them on merge-to-main for this reason.